nixos/modules/system/boot/systemd-nspawn.nix

```diff
@@ -105,29 +171,40 @@ in {
     systemd.nspawn = mkOption {
       default = {};
       type = with types; attrsOf (submodule instanceOptions);
-      description = "Definition of systemd-nspawn configurations.";
+      description = ''
+        Definition of <citerefentry><refentrytitle>systemd.nspawn</refentrytitle>
+        <manvolnum>5</manvolnum></citerefentry> configurations.
+      '';
+    };
+
+    systemd.customContainers = mkOption {
```

Comment:
IIRC the ExecStart override was needed as we encountered issues with unprivileged users in systemd-nspawn in conjunction with `DynamicUser=true` in services, so this should probably be enabled by default.

Also, I'm not even sure if we should add a toggle for this (but if we keep this, I guess that this should be something like `systemd.disableUnprivilegedNspawn`).

Reply:
> Also, I'm not even sure if we should add a toggle for this 

Exactly, it is not a toggle but just a comment out in my fork. If this breaking change is acceptable, removing this toggle is an option (I'm not sure right direction though).

Reply:
changed to disableUnprivilegedNspawn